airflow/contrib/hooks/aws_athena_hook.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from time import sleep
+from airflow.contrib.hooks.aws_hook import AwsHook
+from uuid import uuid4
+
+INTERMEDIATE_STATES = ('QUEUED', 'RUNNING',)
+FINAL_STATES = ('SUCCEEDED', 'FAILED', 'CANCELLED',)
+
+
+class AWSAthenaHook(AwsHook):
+
+    def __init__(self, query, database, region_name, output_location, aws_conn_id=None, *args, **kwargs):
+        self.query = query
+        self.database = database
+        self.region_name = region_name
+        self.outputLocation = output_location
+        self.aws_conn_id = aws_conn_id
+        if 'client_request_token' in kwargs:  # This ensures idempotency of query execution
```

Comment:
We could also simplify this and do it on a single line:
```suggestion
self.client_request_token = kwargs.get('client_request_token') or {}
```